src/provider/utils.ts

```diff
@@ -18,6 +18,42 @@ export function formatHash() {}
  */
 export function txIdentifier() {}
 
+// hex string and BN are detected as block hashes
+// decimal string and number are detected as block numbers
+// null appends nothing to the request url
+export type BlockIdentifier = BlockNumber | BigNumberish;
+type BlockIdentifierObject =
+  | { type: 'BLOCK_NUMBER'; data: BlockNumber }
+  | { type: 'BLOCK_HASH'; data: BigNumberish };
+
+/**
+ * Identifies the block to be queried.
+ *
+ * @param blockIdentifier - block identifier
+ * @returns block identifier object
+ */
+export function getBlockIdentifier(blockIdentifier: BlockIdentifier): BlockIdentifierObject {
```

Comment:
Nothing to change but a comment: I'm not very familiar with all the edge cases. Just hope that we catch all edge cases which seems like there could be a lot or more in the future.

Reply:
I think if we pass in a `BlockIdentifierObject` instead of `BlockIdentifier`, we may not need to handle all edge cases. Thoughts?